### PR TITLE
Orbit Determination

### LINKED ISSUE #10: PNP-Based Orbit Determination Implementation

### PR DESCRIPTION
**Pull request by amaarquadri**
closes #10

Code complete, testing blocked by issue with brahe Epochs

### CODE REVIEW

File: payload/orbit_determination/nonlinear_least_squares_od.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            states = X.reshape(-1, 6)
+            positions = states[:, :3]
+
+            res = np.zeros(3 * (N - 1) + 3 * len(times))
+
+            # dynamics residuals
+            epoch = starting_epoch
+            for i in range(N - 1):
+                res[3 * i:3 * (i + 1)] = positions[i + 1] - self.f(states[i, :], epoch)[:3]
+                epoch = increment_epoch(epoch, self.dt)
```
Comment (Herpderk) [inline payload/orbit_determination/nonlinear_least_squares_od.py:99]: what's `epoch` being used for?

Comment (amaarquadri) [reply payload/orbit_determination/nonlinear_least_squares_od.py:99]: I agree I don't think it should be necessary, but for some reason it's used in the [`gravity model`](https://github.com/cmu-argus-2/GNC-Simulation/blob/7ffc7a37cff94626ce1f5e88b2401cdd44cf2c85/world/physics/models/gravity.py#L32)

Comment (Herpderk) [reply payload/orbit_determination/nonlinear_least_squares_od.py:99]: But you're not using this instance of `epoch` in anything after incrementing it

Comment (Herpderk) [reply payload/orbit_determination/nonlinear_least_squares_od.py:99]: Oh wait nevermind, I see that it's being used in the dynamics

**Comment by Herpderk** [comment]
It looks good! We should wait until the Brahe issue gets resolved and generate some results before merging though

**Pull request merged**